src/ripple/app/rdb/impl/RelationalDBInterface_global.cpp

```diff
@@ -0,0 +1,831 @@
+        // from a different file download.
+        if (pathFromDb != path.string())
+        {
+            session << "DROP TABLE Download;";
+        }
+
+        // Continuing a file download.
+        else
+        {
+            session << "SELECT SUM(LENGTH(Data)) FROM Download;",
+                soci::into(size);
+        }
+    }
+
+    return {std::move(conn), (size ? *size : std::optional<std::uint64_t>())};
+}
+
+std::uint64_t
+databaseBodyDoPut(
+    soci::session& session,
+    std::string const& data,
+    std::string const& path,
+    std::uint64_t fileSize,
+    std::uint64_t part,
+    std::uint16_t maxRowSizePad)
+{
+    std::uint64_t rowSize = 0;
+    soci::indicator rti;
+
+    std::uint64_t remainingInRow = 0;
+
+    auto be =
+        dynamic_cast<soci::sqlite3_session_backend*>(session.get_backend());
+    BOOST_ASSERT(be);
+
+    // This limits how large we can make the blob
+    // in each row. Also subtract a pad value to
+    // account for the other values in the row.
+    auto const blobMaxSize =
+        sqlite_api::sqlite3_limit(be->conn_, SQLITE_LIMIT_LENGTH, -1) -
+        maxRowSizePad;
+
+    std::string newpath;
+
+    auto rowInit = [&] {
+        session << "INSERT INTO Download VALUES (:path, zeroblob(0), 0, :part)",
+            soci::use(newpath), soci::use(part);
+
+        remainingInRow = blobMaxSize;
+        rowSize = 0;
+    };
+
+    session << "SELECT Path,Size,Part FROM Download ORDER BY Part DESC "
+               "LIMIT 1",
+        soci::into(newpath), soci::into(rowSize), soci::into(part, rti);
+
+    if (!session.got_data())
+    {
+        newpath = path;
+        rowInit();
+    }
+    else
+        remainingInRow = blobMaxSize - rowSize;
+
+    auto insert = [&session, &rowSize, &part, &fs = fileSize](
+                      auto const& data) {
+        std::uint64_t updatedSize = rowSize + data.size();
+
+        session << "UPDATE Download SET Data = CAST(Data || :data AS blob), "
+                   "Size = :size WHERE Part = :part;",
+            soci::use(data), soci::use(updatedSize), soci::use(part);
+
+        fs += data.size();
+    };
+
+    size_t currentBase = 0;
+
+    while (currentBase + remainingInRow < data.size())
+    {
+        if (remainingInRow)
+        {
+            insert(data.substr(currentBase, remainingInRow));
+            currentBase += remainingInRow;
+        }
+
+        ++part;
+        rowInit();
+    }
+
+    insert(data.substr(currentBase));
+
+    return part;
+}
+
+void
+databaseBodyFinish(soci::session& session, std::ofstream& fout)
+{
+    soci::rowset<std::string> rs =
+        (session.prepare << "SELECT Data FROM Download ORDER BY PART ASC;");
+
+    // iteration through the resultset:
+    for (auto it = rs.begin(); it != rs.end(); ++it)
+        fout.write(it->data(), it->size());
+}
+
+/* Vacuum DB */
+
+bool
+doVacuumDB(DatabaseCon::Setup const& setup)
+{
+    using namespace boost::filesystem;
+    path dbPath = setup.dataDir / TxDBName;
```

Comment:
The `boost::filesystem::path` is hidden by the declaration of `namespace ripple::path` in src/ripple/app/paths/Types.h and elsewhere.  Replace these two lines with
```
boost::filesystem::path dbPath = setup.dataDir / TxDBName;
```